gui/default/assets/css/overrides.css

```
@@ -189,6 +189,11 @@ input[type="checkbox"].extended-attributes-filter-rule-checkbox {
     word-break: break-all;
 }
 
+/* Break up long words in paragraphs only if necessary to prevent text overflow. */
+.overflow-break-word {
+    overflow-wrap: break-word;
```

```suggestion
    word-wrap: break-word;
```

Can we use `word-wrap` for backwards compatibility? It's the original name of the property and works in every browser possible.

I also thought that maybe it could be possible to replace `word-break: break-all` with `word-wrap: break-word` altogether, but no, it doesn't play nicely with long paths in the Recent Changes modal, pushing the date out of the modal boundaries.

---

Are the two really equivalent though?  From the quoted source, I got the impression that `overflow-wrap` allows breaking long words, while otherwise keeping the "regular" word-wrapping intact.

Do you have an overview of browser (in)compatibility?

---

Mozilla says `overflow-wrap` and the old `word-wrap` are just aliases.

> The property was originally a nonstandard and unprefixed Microsoft extension called word-wrap, and was implemented by most browsers with the same name. It has since been renamed to overflow-wrap, with word-wrap being an alias.

Source: https://developer.mozilla.org/docs/Web/CSS/overflow-wrap

---

https://stackoverflow.com/a/61866696/9294540

---

So MDN lists no browser that doesn't understand `overflow-wrap`, right?  I'll keep the standardized name then, since that's more future-proof and the Right Thing.  If IE11 wraps them in a funky fashion, nothing is lost and people will get a feeling of how old their software stack is.

---

The full support is listed at https://caniuse.com/wordwrap (the MDN is limited in terms of its browser list). Personally, I don't really see a point in using `overflow-wrap` if `word-wrap` is the exact same thing with 100% browser support... Just seems like a property name change for the sake of changing the name.

---

Well, one is CSS and the other is random technical debt.  When writing new code, we shouldn't use stuff that's only kept for compatibility with past mistakes, *if* there is a proper replacement.

---

Both are actually part of the official CSS3 specification:

https://www.w3.org/TR/css-text-3/#overflow-wrap-property

> For legacy reasons, UAs must treat [word-wrap](https://www.w3.org/TR/css-text-3/#propdef-word-wrap) as a [legacy name alias](https://www.w3.org/TR/css-cascade-5/#legacy-name-alias) of the [overflow-wrap](https://www.w3.org/TR/css-text-3/#propdef-overflow-wrap) property.

---

I stand corrected.  Still, what use case exactly do we need to use the legacy name for?  If the non-legacy variant works everywhere, I'd like to avoid introducing ancient "legacy" stuff in 2024.

---

To maintain the same level of backwards compatibility? Especially if, as in this specific case, doing so requires literally zero effort. I mean, it does make sense to use `overflow-wrap` if the rest of the code doesn't support old browsers already, but this isn't the case with the Web GUI.

> I'd like to avoid introducing ancient "legacy" stuff in 2024.

It's the whole Web GUI code "legacy" at this point?

---

Done.